generators/client/templates/react/src/main/webapp/app/modules/administration/audits/audits.tsx.ejs

```diff
@@ -60,119 +48,115 @@ const today = (): string => {
   return toDate.toISOString().slice(0, 10);
 };
 
-export class AuditsPage extends React.Component<IAuditsPageProps, IAuditsPageState> {
-  state: IAuditsPageState = {
-    ...getSortState(this.props.location, ITEMS_PER_PAGE),
-    fromDate: previousMonth(),
-    toDate: today()
+export const AuditsPage = (props: IAuditsPageProps) => {
+  const [pagination, setPagination] = useState(getSortState(props.location, ITEMS_PER_PAGE));
+  const [fromDate, setFromDate] = useState(previousMonth());
+  const [toDate, setToDate] = useState(today());
+
+  useEffect(() => {
+    getAllAudits();
+  }, [fromDate, toDate]);
+
+  useEffect(() => {
+    transition();
+  }, [pagination.activePage, pagination.order, pagination.sort]);
+
+  const onChangeFromDate = evt => setFromDate(evt.target.value);
+
+  const onChangeToDate = evt => setToDate(evt.target.value);
+
+  const sort = p => () =>
+    setPagination({
+      ...pagination,
+      order: pagination.order === 'asc' ? 'desc' : 'asc',
+      sort: p
+    });
+
+  const transition = () => {
+    getAllAudits();
```

Comment:
remove the `getAllAudits()` here and move pagination data to the first `useEffect()` in order to avoid extra api call